src/Utilities/TupleSlice.hpp

```diff
@@ -0,0 +1,55 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines functions for slicing tuples
+
+#pragma once
+
+#include <cstddef>
+#include <tuple>
+#include <utility>
+
+namespace tuple_impl_detail {
+
+template <size_t Start, typename Tuple, size_t... Is>
+constexpr auto slice_impl(Tuple&& tuple,
+                          std::index_sequence<Is...> /*meta*/) noexcept {
+  return std::forward_as_tuple(
+      std::get<Is + Start>(std::forward<Tuple>(tuple))...);
+}
+
+}  // namespace tuple_impl_detail
+
+/// \ingroup UtilitiesGroup
+/// References to the subset of elements in `tuple` from index `Start` to
+/// (excluding) `Stop`
```

Comment:
Either give an explicit return type or make the documentation more explicit about the result.  I assumed `tuple_slice<0, 1>(std::declval<const std::tuple<int, double>&>())` would have type `std::tuple<int>`, but this mentions references.

Reply:
The idea here is taken from https://ngathanasiou.wordpress.com/2016/11/28/want-a-std-slice/. The `tuple_slice` references lvalues and move-constructs rvalues. This also allows assigning to a slice, for instance, as well als forwarding a subset of arguments to a function: https://gist.github.com/picanumber/bba2339c5648b53355487cd97d2b70f7#file-slice_examples-cpp
The implementation with `std::forward_as_tuple` is very straight-forward. I'd be interested to hear yours and @nilsdeppe opinion on this!

Reply:
Ah, OK, I see.  This needs a lot more documentation, then.  That's not what I thought it was trying to do.  I'm also still not sure that I know what the return type will be if I pass it a temporary.